### PR TITLE
Update to a ready for demoing state

### PR DESCRIPTION
**Pull request by jgw96**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,7 @@ platforms/
 plugins/
 plugins/android.json
 plugins/ios.json
+www/
```
Comment (brandyscarney) [inline .gitignore]: why was this added to the .gitignore? This folder needs to be added to the repo in order to start an app with the conference app without errors, e.g. `ionic start --v2 myApp conference`

File: package.json [hunk 2]
```diff
@@ -34,9 +34,12 @@
     "zone.js": "0.6.21"
   },
   "devDependencies": {
-    "@ionic/app-scripts": "0.0.39",
+    "@ionic/app-scripts": "0.0.41",
     "typescript": "^2.0.3"
   },
+  "config": {
+    "ionic_bundler": "webpack"
```
Comment (brandyscarney) [inline package.json]: Why is this needed? Are we just adding it to show an example of how to set it?

Comment (jgw96) [inline package.json]: Actually, this should be set to rollup to get the slightly smaller and faster bundle that it provides.

Comment (jgw96) [inline package.json]: Also provides a nice example of how to use that config option.

**Comment by brandyscarney** [comment]
The `Show Tutorial` button is only displayed when you aren't logged in, as soon as you log in the button disappears from the menu. It needs to be added to the other menu or as part of the array of navigation pages: https://github.com/driftyco/ionic-conference-app/blob/master/src/app/app.template.html#L37-L69

File: package.json [hunk 3]
```diff
@@ -34,9 +34,12 @@
     "zone.js": "0.6.21"
   },
   "devDependencies": {
-    "@ionic/app-scripts": "0.0.39",
+    "@ionic/app-scripts": "0.0.41",
     "typescript": "^2.0.3"
   },
+  "config": {
+    "ionic_bundler": "rollup"
```
Comment (brandyscarney) [inline package.json]: Why did we switch to rollup?

File: src/pages/about-popover/about-popover.ts [hunk 4]
```diff
@@ -1,16 +1,37 @@
...
 
-  constructor(public viewCtrl: ViewController) { }
+  constructor(
+    public viewCtrl: ViewController,
+    public navCtrl: NavController,
+    public app: App,
+    public modalCtrl: ModalController
+  ) { }
+
+  support() {
```
Comment (brandyscarney) [inline src/pages/about-popover/about-popover.ts]: We should add Support as an Account link. I didn't even know this page existed until I was going through the code.

File: src/app/app.component.ts [hunk 5]
```diff
@@ -45,21 +46,35 @@ export class ConferenceApp {
...
       StatusBar.styleDefault();
       Splashscreen.hide();
     });
 
+    // Check if the user has already seen the tutorial
+    this.userData.checkHasSeenTutorial().then((hasSeenTutorial) => {
+      if (hasSeenTutorial === null) {
+        // User has not seen tutorial
+        this.rootPage = TutorialPage;
+        this.storage.set('hasSeenTutorial', 'true');
```
Comment (brandyscarney) [inline src/app/app.component.ts]: I don't think we should set this to `true` until the user presses "Skip" or "Continue" from the tutorial. If I launch the app on my phone and then exit the app without interacting with it the next time I open it the tutorial will be skipped. Then I have to open the menu to see it again when I never really _said_ I saw it.

File: package.json (hunk 3 above)
Comment (jgw96) [inline package.json]: For the perf advantages. Rollup gives us a slightly smaller bundle than webpack does and also does not use closures in its bundle like webpack does. JS engines cannot easily optimize code in closures, which leads to slower code execution.

File: src/pages/about-popover/about-popover.ts (hunk 4 above)
Comment (jgw96) [inline src/pages/about-popover/about-popover.ts]: 👍  can do. The support page can be reached from the popover on the about page.

File: src/app/app.component.ts (hunk 5 above)
Comment (jgw96) [inline src/app/app.component.ts]: 👍  on this. Ill implement.

File: src/app/app.module.ts [hunk 6]
```diff
@@ -18,10 +18,24 @@ import { SpeakerDetailPage } from '../pages/speaker-detail/speaker-detail';
...
 import { TutorialPage } from '../pages/tutorial/tutorial';
+import { Support } from '../pages/support/support';
 
 import { ConferenceData } from '../providers/conference-data';
 import { UserData } from '../providers/user-data';
 
+export const deepLinkConfig: DeepLinkConfig = {
+  links: [
+    { component: MapPage, name: 'map', segment: 'map' },
+    { component: SpeakerListPage, name: 'speakerList', segment: 'speaker-list' },
```
Comment (brandyscarney) [inline src/app/app.module.ts]: Why is the name for `speakerList` camelCase when the others are not? Can these be consistent.

File: src/app/app.module.ts [hunk 7]
```diff
@@ -18,10 +18,24 @@ import { SpeakerDetailPage } from '../pages/speaker-detail/speaker-detail';
...
+    { component: AboutPage, name: 'about', segment: 'about' },
+    { component: Support, name: 'support', segment: 'support' },
+    { component: SignupPage, name: 'signup', segment: 'signup' },
```
Comment (brandyscarney) [inline src/app/app.module.ts]: Need to add the AccountPage in here.

File: src/app/app.module.ts (hunk 6 above)
Comment (jgw96) [inline src/app/app.module.ts]: 👍  will fix.

**Comment by brandyscarney** [comment]
There is something weird going on with Deeplink. If I go to the following URL `http://localhost:8100/#/about/about` it loads the About page for about a second and then redirects home.

File: src/app/app.module.ts [hunk 8]
```diff
@@ -18,10 +18,24 @@ import { SpeakerDetailPage } from '../pages/speaker-detail/speaker-detail';
...
+    { component: LoginPage, name: 'login', segment: 'login' },
+    { component: TutorialPage, name: 'tutorial', segment: 'tutorial' },
+    { component: SessionDetailPage, name: 'session-detail', segment: 'session-detail' },
+    { component: SpeakerDetailPage, name: 'speaker-detail', segment: 'speaker-detail' }
```
Comment (jgw96) [inline src/app/app.module.ts]: 👍  good idea

**Comment by jgw96** [comment]
Hmmm, that almost sounds like a deeplink bug. Gonna try using the deeplinker with just a plain starter and see if i can repro.

**Comment by brandyscarney** [comment]
Well changing this code:

```

    // Check if the user has already seen the tutorial
    this.userData.checkHasSeenTutorial().then((hasSeenTutorial) => {
      if (hasSeenTutorial === null) {
        // User has not seen tutorial
        this.rootPage = TutorialPage;
        this.storage.set('hasSeenTutorial', 'true');
      } else {
        // User has seen tutorial
        this.rootPage = TabsPage;
      }
    });
```

to just this:

```
    this.rootPage = TabsPage;
```

fixes it.

**Comment by biesbjerg** [comment]
Regarding the deeplinking, I created an issue here: https://github.com/driftyco/ionic/issues/9074

File: src/app/app.component.ts (hunk 5 above)
Comment (jgw96) [inline src/app/app.component.ts]: Just pushed a commit to fix this.

File: src/app/app.component.ts [hunk 9]
```diff
@@ -1,7 +1,8 @@
 import { Component, ViewChild } from '@angular/core';
 
-import { Events, MenuController, Nav, Platform } from 'ionic-angular';
+import { Events, MenuController, Nav, Platform, ToastController } from 'ionic-angular';
```
Comment (brandyscarney) [inline src/app/app.component.ts]: Let's remove ToastController since it isn't used

File: src/pages/map/map.ts [hunk 10]
```diff
@@ -2,44 +2,81 @@ import { Component } from '@angular/core';
...
   selector: 'page-map',
   templateUrl: 'map.html'
 })
 export class MapPage {
-  constructor(public confData: ConferenceData) {}
+
+  public map: GoogleMap;
+
+  constructor(public confData: ConferenceData, public platform: Platform) {
+    // this.platform.ready().then(() => this.onPlatformReady());
```
Comment (brandyscarney) [inline src/pages/map/map.ts]: Remove commented out code

File: src/pages/schedule/schedule.ts [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import { Component, ViewChild } from '@angular/core';
 
-import { AlertController, App, ItemSliding, List, ModalController, NavController } from 'ionic-angular';
+import { AlertController, App, ItemSliding, List, ModalController, NavController, ToastController } from 'ionic-angular';
```
Comment (brandyscarney) [inline src/pages/schedule/schedule.ts]: Remove ToastController, unused

File: src/pages/schedule/schedule.ts [hunk 12]
```diff
@@ -85,7 +83,10 @@ export class SchedulePage {
   goToSessionDetail(sessionData) {
     // go to the session detail page
     // and pass in the session data
-    this.navCtrl.push(SessionDetailPage, sessionData);
+    this.navCtrl.push(SessionDetailPage, {
+      data: sessionData,
+      session: sessionData.name
```
Comment (brandyscarney) [inline src/pages/schedule/schedule.ts]: Why are we passing name in a separate property? This is already passed as part of sessionData.

Comment (jgw96) [inline src/pages/schedule/schedule.ts]: This was part of the changes needed for adding the deeplinker. Id be in favor of keeping this change for now so that when deeplinker is ready and we add it again I can just drop in the config.

**Comment by brandyscarney** [comment]
Looks good 👍
